### PR TITLE
chore: Update dependencies

### PR DESCRIPTION
**Pull request by BulkBeing**
- Ensures the serving demo works from current main of numaflow.
- Updated `numaflow-go` and`numaflow-rs` to their current main.

### CODE REVIEW

File: serving-demo/manifests/pipeline.yaml [hunk 1]
```diff
@@ -20,70 +22,91 @@ spec:
...
     - name: error-sink
+      containerTemplate:
+        imagePullPolicy: Never
       scale:
         min: 1
       sink:
         udsink:
           container:
             image: servesink:0.1
+            imagePullPolicy: Never
```
Comment (vigith) [inline serving-demo/manifests/pipeline.yaml:109]: Should we set it to Always, since this is experimental?

Comment (BulkBeing) [reply serving-demo/manifests/pipeline.yaml:109]: In this example we want to use the image we build locally. We don't want the image to be pulled from a container registry right?

Comment (vigith) [reply serving-demo/manifests/pipeline.yaml:109]: as per the readme, we are doing local import

Comment (BulkBeing) [reply serving-demo/manifests/pipeline.yaml:109]: Yes. So it shouldn't be pulled from the container registry. Instead, use what's available locally

**Review by vigith — APPROVED** [review]

**Pull request merged**